Mail/Components/ThreadCell/ThreadCell.swift

```diff
@@ -60,13 +60,24 @@ struct ThreadCellDataHolder {
 
         isInWrittenByMeFolder = FolderRole.writtenByMeFolders.contains { $0 == thread.folder?.role }
 
-        let content: String?
+        let contentMessage: Message?
         if isInWrittenByMeFolder {
             recipientToDisplay = lastMessageNotFromSent?.to.first
-            content = (thread.lastMessageFromFolder ?? thread.messages.last)?.preview
+            contentMessage = thread.lastMessageFromFolder ?? thread.messages.last
         } else {
             recipientToDisplay = lastMessageNotFromSent?.from.first
-            content = thread.messages.last?.preview
+            contentMessage = thread.messages.last
+        }
+
+        let content: String?
+        if let message = contentMessage {
+            if message.encrypted {
+                content = MailResourcesStrings.Localizable.encryptedMessageHeader
+            } else {
+                content = message.preview
+            }
+        } else {
+            content = nil
         }
 
         if let content, !content.isEmpty {
```

Comment:
We don't need this part anymore. You can merge the logic directly in your new block.

Instead of assigning content, assign preview.